src/ripple/rpc/impl/TransactionSign.cpp

```diff
@@ -447,9 +447,20 @@ transactionPreProcessImpl (
             tx_json[jss::Flags] = tfFullyCanonicalSig;
     }
 
-    // If multisigning then we need to return the public key.
+    // If multisigning there should not be a single signature and vice versa.
     if (signingArgs.isMultiSigning())
+    {
+        if (tx_json.isMember (jss::TxnSignature))
+            return rpcError (rpcALREADY_SINGLE_SIG);
+
+        // If multisigning then we need to return the public key.
         signingArgs.setPublicKey (keypair.first);
+    }
+    else
+    {
+        if (tx_json.isMember (sfSigners.jsonName))
```

Comment:
Can we start on the path to consistency? Add `jss:Signers` to `JsonFields.h` and use it here.

How about adding a `signingArgs.isSingleSigning` method, using it for this branch, and adding a new else-branch that diagnoses an error ("unknown signing method").

Reply:
I started looking at the consistency problem.  I think it deserves its own pull request that really does the whole job.  Otherwise all we accomplish is stirring mud.  And my personal preference would be that the `SField` objects would not be represented in the `jss` namespace at all.  All `SField` names should be retrieved from their primary source.

I'll put together a separate pull request with that change.

I'll add an `isSingleSigning()` method.  That does help with documentation a bit.  I can't bring myself to justify a third branch to the if-else.  Adding a third kind of signing will be a big deal.  The person adding a third kind of signing will need to add a lot more than a third condition in the if-else.